project_generator/tools/uvision.py

```diff
@@ -69,6 +71,10 @@ class uVisionDefinitions():
             'Utilities': {
                 'Flash2': 'STLink\\ST-LINKIII-KEIL_SWO.dll',
             },
+            'OptxFile' : {
```

Comment:
I don't have the possibility to check if it is a common one: I don't have Keil4.

For me the new principle looks logical. I was not very satisfied with my first proposal but didn't had a better idea.